src/storage/page_buffer.c

```diff
@@ -8408,7 +8421,7 @@ pgbuf_respond_data_fetch_page_request (THREAD_ENTRY &thread_r, std::string &payl
     }
 
   int error = NO_ERROR;
-  PAGE_PTR page_ptr = pgbuf_fix (&thread_r, &vpid, OLD_PAGE, PGBUF_LATCH_READ, PGBUF_UNCONDITIONAL_LATCH);
+  PAGE_PTR page_ptr = pgbuf_fix (&thread_r, &vpid, fetch_mode, PGBUF_LATCH_READ, PGBUF_UNCONDITIONAL_LATCH);
```

Comment:
Adapt behavior according to new possbile state:
- if in recovery and page is null and no error is set, return empty response - make sure to propagate correct state on the active transaction server side in `pgbuf_request_data_page_from_page_server`
- in in recover and page is null and error is set, respond with error
- happy path currently handled on `else`